server/events/command_requirement_handler.go

```diff
@@ -17,6 +18,32 @@ type DefaultCommandRequirementHandler struct {
 	WorkingDir WorkingDir
 }
 
+func (a *DefaultCommandRequirementHandler) ValidatePlanProject(repoDir string, ctx command.ProjectContext) (failure string, err error) {
+	for _, req := range ctx.PlanRequirements {
+		switch req {
+		case raw.ApprovedRequirement:
+			if !ctx.PullReqStatus.ApprovalStatus.IsApproved {
+				return "Pull request must be approved by at least one person other than the author before running plan.", nil
+			}
+		// this should come before mergeability check since mergeability is a superset of this check.
+		case valid.PoliciesPassedCommandReq:
```

Comment:
I'm considering this should be set or not.

This feature is useful for running policy check before plan. It  ensure not to run invalid terraform running.
Currently this field is set by `--enable-policy-checks` user config. For all users who use `--enable-policy-checks`, they requires policy check before plan by default. This become a breaking change.

https://github.com/runatlantis/atlantis/blob/cb0aadf57127d67223483b7ab2184df8b2671a5f/server/core/config/valid/global_cfg.go#L225-L227


I think `--enable-policy-checks` affect `ApplyRequirements` only and `PlanRequirements` use `plan_requirements` fields only.

cc @runatlantis/maintainers

Reply:
And this requires ctx.ProjectPlanStatus, so something is strange?

Reply:
I just propose removing policy check for plan requirement and it should be considered in the another issue or PR.

Reply:
I've removed the policy check and policy check test for plan